connector_helpscout/models/helpscout_conversation/importer.py

```diff
@@ -0,0 +1,131 @@
+              (none('subject'), 'name'),
+              ('created_at', 'backend_date_created'),
+              ('modified_at', 'backend_date_modified'),
+              ]
+
+    @mapping
+    @only_create
+    def odoo_id(self, record):
+        conversation = self.env['project.task'].search([
+            ('code', '=', str(record.number)),
+        ])
+
+        if conversation:
+            return {'odoo_id': conversation.id}
+
+    @mapping
+    def helpscout_folder_id(self, record):
+        folder_external_id = "%d,%d" % (
+            record.mailbox.id,
+            record.folder_id,
+        )
+        folder_binding = self.env['helpscout.helpscout.folder'].search([
+            ('external_id', '=', folder_external_id)
+        ])
+        folder = folder_binding.odoo_id
+        if folder:
+            return {'helpscout_folder_id': folder.id}
+
+    @mapping
+    def partner_id(self, record):
+        customer = self.env['helpscout.customer'].search([
+            ('external_id', '=', record.customer.id)
+        ])
+        partner = customer.odoo_id
+        return {'partner_id': partner.id}
+
+    @mapping
+    def project_id(self, record):
+        mailbox = self.env['helpscout.mailbox'].search([
+            ('external_id', '=', record.mailbox.id)
+        ])
+        project = mailbox.odoo_id
+        if project:
+            return {'project_id': project.id}
+
+    @mapping
+    def stage_id(self, record):
+        stage = self.env.ref('connector_helpscout.status_%s' % record.status)
+        if stage:
+            return {'stage_id': stage.id}
+
+    @mapping
+    def tag_ids(self, record):
+        tags = self.env['project.tags'].search([
+            ('name', 'in', record.tags)
+        ])
+        if tags:
+            tag_ids = tags.mapped('id')
+            return {'tag_ids': [(6, 0, tag_ids)]}
+
+    @mapping
+    def user_id(self, record):
+        try:
+            owner = self.env['helpscout.user'].search([
+                ('external_id', '=', record.owner.id)
+            ])
+        except AttributeError:
+            # in case a user/owner has not been assigned to the ticket
+            return {'user_id': False}
+        user = owner.odoo_id
+        return {'user_id': user.id}
+
+
+class HelpScoutConversationImporter(Component):
+    """Import one HelpScout record."""
+    _name = 'helpscout.record.importer.conversation'
+    _inherit = 'helpscout.importer'
+    _apply_on = 'helpscout.conversation'
+
+    def _import_dependencies(self):
+        """
+        Import dependencies for HelpScout conversation.
+
+        Folders are found using the Mailbox id, not the folder id.
+        Tags can only be retrieved as a complete list.
+        """
+        self.env['helpscout.helpscout.folder'].import_batch(
+            self.backend_record,
+            {'mailbox_id': self.helpscout_record['mailbox']['id']},
+        )
+        self.env['helpscout.tag'].import_batch(self.backend_record)
```

Comment:
Hmmm now that I think about it, instead of doing this every time (tags probably won't change much), we can save a lot of API calls by doing it in the tag_ids mapping method instead.

We should search for pre-existing tags, then compare against the amount of tags that actually exist in the record. If they do not match, call `import_batch`, then return the recursive value:

```python
@mapping
def tag_ids(self, record):
    tags = self.env['helpscout.project.tags'].search([..])
    if len(tags) != len(record.tags):
        self.env['helpscout.project.tags'].import_batch(backend)
        return self.tag_ids(record)
    return {'tag_ids': [(6, 0, tags.ids)]}
```